### PR TITLE
Use Optional to represent missing LIKE escape character

### PR DESCRIPTION
**Pull request by martint**
Extracted from https://github.com/prestodb/presto/pull/10816 with comments addressed.

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java [hunk 1]
```diff
@@ -622,9 +622,10 @@ protected Type visitLikePredicate(LikePredicate node, StackableAstVisitorContext
 
             Type patternType = getVarcharType(node.getPattern(), context);
             coerceType(context, node.getPattern(), patternType, "Pattern for LIKE expression");
-            if (node.getEscape() != null) {
-                Type escapeType = getVarcharType(node.getEscape(), context);
-                coerceType(context, node.getEscape(), escapeType, "Escape for LIKE expression");
+            if (node.getEscape().isPresent()) {
+                Expression escape = node.getEscape().get();
+                Type escapeType = getVarcharType(escape, context);
+                coerceType(context, escape, escapeType, "Escape for LIKE expression");
```
Comment (kokosing) [inline presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java]:
```
node.getEscape().ifPresent(escape -> coerceType(context, escape, getVarcharType(escape, context), "Escape for LIKE expression"))
```
?

File: presto-parser/src/main/java/com/facebook/presto/sql/ExpressionFormatter.java [hunk 2]
```diff
@@ -512,10 +512,12 @@ protected String visitLikePredicate(LikePredicate node, Void context)
...
 
-            if (node.getEscape() != null) {
-                builder.append(" ESCAPE ")
-                        .append(process(node.getEscape(), context));
-            }
+            node.getEscape().ifPresent(escape -> {
+                if (node.getEscape().isPresent()) {
+                    builder.append(" ESCAPE ")
+                            .append(process(node.getEscape().get(), context));
+                }
```
Comment (kokosing) [inline presto-parser/src/main/java/com/facebook/presto/sql/ExpressionFormatter.java]:
```
node.getEscape().ifPresent(escape -> builder.append(" ESCAPE ").append(process(escape), context)));
```
?

File: presto-parser/src/main/java/com/facebook/presto/sql/ExpressionFormatter.java [hunk 3]
```diff
@@ -512,10 +512,12 @@ protected String visitLikePredicate(LikePredicate node, Void context)
                     .append(" LIKE ")
                     .append(process(node.getPattern(), context));
 
-            if (node.getEscape() != null) {
-                builder.append(" ESCAPE ")
-                        .append(process(node.getEscape(), context));
-            }
+            node.getEscape().ifPresent(escape -> {
+                if (node.getEscape().isPresent()) {
```
Comment (kokosing) [inline presto-parser/src/main/java/com/facebook/presto/sql/ExpressionFormatter.java]: it is already checked

File: presto-parser/src/main/java/com/facebook/presto/sql/tree/DefaultTraversalVisitor.java [hunk 4]
```diff
@@ -330,8 +330,8 @@ protected R visitLikePredicate(LikePredicate node, C context)
     {
         process(node.getValue(), context);
         process(node.getPattern(), context);
-        if (node.getEscape() != null) {
-            process(node.getEscape(), context);
+        if (node.getEscape().isPresent()) {
```
Comment (kokosing) [inline presto-parser/src/main/java/com/facebook/presto/sql/tree/DefaultTraversalVisitor.java]: use `ifPresent` here

File: presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java (hunk 1 above)
Comment (martint) [reply presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java]: I think that makes the code less readable. That single line ends up doing too much and makes it harder to spot that there's a side-effect implied by ifPresent

**Pull request closed without merging**